src/writers/sink.creator.hh

```diff
@@ -0,0 +1,95 @@
+#ifndef H_ACQUIRE_STORAGE_ZARR_WRITERS_SINK_CREATOR_V0
+#define H_ACQUIRE_STORAGE_ZARR_WRITERS_SINK_CREATOR_V0
+
+#include "sink.hh"
+#include "../common.hh"
+
+#include <optional>
+#include <memory>
+
+namespace acquire::sink::zarr {
+struct SinkCreator final
+{
+  public:
+    SinkCreator() = delete;
```

Comment:
> Wondering why this is necessary? Providing any constructor in a C++ class suppresses the implicit generation of the default constructor by the compiler.

In these cases I prefer to be explicit. Future contributors should not try to implement a `SinkCreator` without a thread pool to work with.

> I'm also curious as to why it's necessary to explicitly state that we're using the default destructor and marking it noexcept.

Not sure at this point, but there might at one point have been a nontrivial destructor, and instead of just removing it from the header I used `default` instead.